src/js/data/index.ts

```diff
@@ -0,0 +1,96 @@
+import { Observable, of, throwError, timer } from "rxjs";
```

Comment:
Having the [reactive-graphql](https://github.com/mesosphere/reactive-graphql) part of the client embedded in DCOS UI would be perfectly fine. 
We could have a dedicated space for it, like `data/cosmos/index.ts` because `data/index.ts` would be the entry point, right?

Reply:
I know @DanielMSchmidt has been working on extension kid for data-layer, but for now what would be the best way to compose different parts of the data-layer? 

For the implementation would we make 2 `graphQLObservable`s, one for the cosmos query and another for the ui-update-service? I could in many ways it would make sense for that to be one query/`graphQLObservable` instead of two. But if there isn't a great way to combine these yet then would 2 be the correct approach?

Reply:
> but for now what would be the best way to compose different parts of the data-layer

The way you would combine those is by passing the type definitions and the schema as an array to the `makeExecutableSchema` call. E.g. You could have a schema describing Nodes and one describing Networks that adds a field to the nodes by passing both into `makeExecutableSchema` with a `{ Nodes: { network(parent, args) {...} }}` field resolver and having a schema like this:

Node Schema:
```graphql
type Node {
  name: String!
}
```

Network Schema:
```graphql
type Network {
   ip: String!
   ipv6: Boolean!
}

extend type Node {
  network: Network!
}
```

This way you can compose smaller schemas into a single bigger one, which is the underlying mechanism we are going to use when we go with the extension-kid.


If I understand your use-case correctly you would like to expose a root level query for searching a single package, which is a good choice IMHO. And now you would like to have the data layer answer the question if there is any update and to which version, am I right?

I could see three ways to solve this:

1) You could have a new root level query `UICanBeUpdated: Boolean!` which uses the logic you provide in your cosmos API to determine if an update can be done. And you probably would do an `updateUi(Version: String!): Boolean!` mutation that performs the update
2) You could use a "nested DataLayer" approach and use the data from the cosmos API in the query and mutation. I'm unsure of the performance impact of this pattern, but I have thought about using it in general for a while now 🤔 
3) You could write one mutation `updateUi(dryRun: Boolean!, version: String!): String!` that has a dry-run functionality, not actually doing the change, but just answering the question to which version it would update. I have read about this pattern in the past and the author seemed to be convinced of it, but I never tried it out

Reply:
Does this (still) yield some action item? 

If so, could someone summarise the comments above to something that's digestible for me?
If not, please resolve!